src/skydb.ts

```diff
@@ -400,3 +525,63 @@ export function getNextRevisionFromEntry(entry: RegistryEntry | null): bigint {
 
   return revision;
 }
+
+/**
+ * Checks whether the raw data link matches the cached data link, if provided.
+ *
+ * @param rawDataLink - The raw, unformatted data link.
+ * @param cachedDataLink - The cached data link, if provided.
+ * @returns - Whether the cached data link is a match.
+ */
+export function checkCachedDataLink(rawDataLink: string, cachedDataLink?: string): boolean {
+  // Double exclamation mark is necessary to convert cachedDataLink to false if it is undefined.
+  return !!(cachedDataLink && rawDataLink === parseSkylink(cachedDataLink));
+}
+
+/**
+ * Gets and validates the registry entry.
+ *
+ * @param client - The Skynet Client
+ * @param publicKey - The user public key.
+ * @param dataKey - The key of the data to fetch for the given user.
+ * @param opts - Additional settings.
+ * @returns - The registry entry, or null if not found or deleted.
+ * @throws - Will throw if the returned entry data is not a valid byte array.
+ */
+async function getRegistryEntry(
```

Comment:
why is this a separate function when we have `client.registry.getEntry` ?

Reply:
We have to check if the returned entry is an empty skylink as that has a specific meaning in the context of SkyDB. I tried renaming the function.